lib/iam_ssoe_oauth/session_manager.rb

```diff
@@ -15,8 +15,34 @@ def find_or_create_user
       create_user_session
     end
 
+    def logout
+      response = iam_ssoe_service.post_revoke(@access_token)
+      destroy_user_session if response.status == 200
+    end
+
     private
 
+    def destroy_user_session
+      uuid = @session.uuid
+
+      identity_destroy_count = IAMUserIdentity.find(uuid).destroy
+      user_destroy_count = IAMUser.find(uuid).destroy
+      session_destroy_count = @session.destroy
+
+      # redis returns number of records successfully deleted
+      if [identity_destroy_count, user_destroy_count, session_destroy_count].all? { |c| c != 0 }
+        Rails.logger.info('IAMUser log out success', uuid: uuid)
+        true
+      else
+        Rails.logger.warn('IAMUser log out failure', uuid: uuid, status: {
```

Comment:
Should the api consumer make an effort not to call /logout unless they think a session is in-progress? Looks like the only side-effect is this warning log message.